### PR TITLE
Password generator

### LINKED ISSUE #77: Password generator popup
Add the password generator tool to the extension when registering, for instance.

### PR DESCRIPTION
**Pull request by perry-mitchell**
Add the password generator. Fixes #77.

### CODE REVIEW

File: source/background/library/contextMenu.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    chrome.contextMenus.create({
+        parentId: __menu,
+        type: "separator",
+        ...CONTEXT_SHARED_ALL
+    });
+    chrome.contextMenus.create({
+        title: "Log in using",
+        parentId: __menu,
+        onclick: () => {
+            // todo
```
Comment (sallar) [inline source/background/library/contextMenu.js]: Are we going to release with these in to-do state?

File: source/dialog/library/messaging.js [hunk 2]
```diff
@@ -34,6 +34,16 @@ export function searchEntriesForURL(url) {
     chrome.runtime.sendMessage({ type: "search-entries-for-url", url });
 }
 
+export function setGeneratedPassword(password) {
+    // chrome.runtime.sendMessage({ type: "set-generated-password", password });
```
Comment (sallar) [inline source/dialog/library/messaging.js]: extra comment here

File: source/tab/generator.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+export function watchInputs() {
+    if (__stopTracking) {
+        stopTrackingInputs();
+    }
+    const searchInputs = debounce(() => {
+        [...document.body.getElementsByTagName("input")].forEach(input => trackInput(input));
+    }, 200);
+    // watch for new ones
+    __stopTracking = mucus(document.body, function(changes) {
+        // changes.added.filter(item => item.tagName.toLowerCase() === "input").forEach(input => trackInput(input));
```
Comment (sallar) [inline source/tab/generator.js]: comment here too

File: source/background/library/contextMenu.js (hunk 1 above)
Comment (perry-mitchell) [reply source/background/library/contextMenu.js]: Ah yeah good point! I completely forgot. I'll put this back into WIP status until I've added them.

Comment (perry-mitchell) [reply source/background/library/contextMenu.js]: Actually, I might hide them until next time as they're a bit of work, let's see..

Comment (sallar) [reply source/background/library/contextMenu.js]: Awesome 👍

**Comment by perry-mitchell** [comment]
@sallar Added an 'about' menu and disabled the others temporarily so we can do those in a separate MR:

image

**Comment by perry-mitchell** [comment]
Back to you @sallar - Cheers

**Pull request merged**